src/importsource.cpp

```diff
@@ -60,15 +60,16 @@ void ImportSource::setUrl(const std::string &url)
 
 ModelPtr ImportSource::model() const
 {
+    if (mPimpl->mModel.expired()) {
+        return nullptr;
+    }
     return mPimpl->mModel.lock();
 }
 
 void ImportSource::setModel(const ModelPtr &model)
 {
-    if (model == nullptr) {
-        if (!mPimpl->mModel.expired()) {
-            mPimpl->mModel.reset();
-        }
+    if ((model == nullptr)&&(!mPimpl->mModel.expired())) {
```

Comment:
1. ClangTidy is not going to like this.
2. I would replace this with `if ((model == nullptr) && !mPimpl->mModel.expired()) {`.
3. What is `model` is equal to `nullptr` and there is no prior model? `mPimp->mModel = model` will be executed and fail...?

Reply:
Isn't there always a mModel there? It's present but .expired() is true? Otherwise we need to change stuff elsewhere too ...

Reply:
I thought you wanted `mPimpl->mModel` to be unset in some cases?

Reply:
Yes I do.  But the way I thought it worked (probably wrong!) is that the pointer always exists, but is set to "expired" when not in use, or a real value otherwise?  And that "reset" sets the "expired" flag to true? IOW, if the given model is null, then reset the pointer so that expired is true.  If the current model pointer is already expired, then we don't need to do anything?

Reply:
Turns out we don't need to test for the mModel == nullptr, it's handled internally somewhere.  I've updated to suit ... and as a bonus, the tests pass now ;)

Reply:
Doh, of course. Much better this way. (I take it you are talking about commit c90cbe2.)